Regression: generated d.ts files for ESM don’t have file extensions

### Describe the bug

The bug reported in #1087 appears again with the latest version `1.50.1` for `@tanstack/react-router` as well as `@tanstack/start`

`react-router`:
```
node_modules/@tanstack/react-router/dist/esm/fileRoute.d.ts:54:31 - error TS2835: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './useNavigate.js'?

54     useNavigate: () => import('./useNavigate').UseNavigateResult<string>;
                                 ~~~~~~~~~~~~~~~

node_modules/@tanstack/react-router/dist/esm/index.d.ts:58:118 - error TS2835: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './utils.js'?

58 pick, functionalUpdate, replaceEqualDeep, isPlainObject, isPlainArray, deepEqual, useStableCallback, shallow, } from './utils';
                                                                                                                        ~~~~~~~~~

node_modules/@tanstack/react-router/dist/esm/link.d.ts:1:43 - error TS2307: Cannot find module '.' or its corresponding type declarations.

1 import { AnyRouter, ParsedLocation } from '.';
```

`start`:
```
node_modules/@tanstack/start/dist/esm/client/index.d.ts:8:70 - error TS2835: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean '../constants.js'?

8 export { serverFnPayloadTypeHeader, serverFnReturnTypeHeader, } from '../constants';
```

### Your Example Website or App

https://stackblitz.com/edit/github-yk78pg-anfycf?file=README.md

### Steps to Reproduce the Bug or Issue

1. set `moduleResolution` in `tsconfig` set to `node16`
1.  import something from `@tanstack/react-router` or `@tanstack/start`
1. run the typescript compiler


### Expected behavior

I would expect the file extensions in the .d.ts file to be present.

### Screenshots or Videos

_No response_

### Platform

- OS: macos
- Version: `1.50.1`


### Additional context

_No response_

--- a/package.json
+++ b/package.json
@@ -33,7 +33,7 @@
     "@eslint-react/eslint-plugin": "^1.8.2",
     "@playwright/test": "^1.45.3",
     "@rollup/plugin-replace": "^5.0.7",
-    "@tanstack/config": "^0.11.1",
+    "@tanstack/config": "^0.11.3",
     "@types/node": "^20.14.7",
     "@types/react": "^18.3.3",
     "@types/react-dom": "^18.3.0",

